test/parallel/test-net-dns-custom-lookup.js

```diff
@@ -65,3 +66,44 @@ check(4, function() {
     }
   }).on('error', common.expectsError({ code: 'ERR_INVALID_IP_ADDRESS' }));
 }
+
+const brokenCustomLookup = (_hostname, options, callback) => {
+  // Incorrectly return an array of IPs instead of a string.
+  callback(null, ['127.0.0.1'], options.family);
+};
+
+describe('when family is ipv4', () => {
+  it('net.connect() throws when lookup does not return a string', (t, done) => {
```

Comment:
I think it would be better to not mix tests run via `node:test` and without it. I would personally move these new tests in a separate file.

Reply:
Good point. I agree.

Reply:
Done <3